src/main/java/org/cryptomator/ui/mainwindow/VaultDetailUnlockedController.java

```diff
@@ -3,38 +3,89 @@
+import java.awt.Desktop;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.Optional;
+import java.util.ResourceBundle;
+import java.util.Set;
+import java.util.stream.Collectors;
 
 @MainWindowScoped
 public class VaultDetailUnlockedController implements FxController {
 
+	private static final Logger LOG = LoggerFactory.getLogger(VaultDetailUnlockedController.class);
+	private static final String ACTIVE_CLASS = "active";
 	private final ReadOnlyObjectProperty<Vault> vault;
 	private final FxApplicationWindows appWindows;
 	private final VaultService vaultService;
 	private final Stage mainWindow;
+	private final ResourceBundle resourceBundle;
 	private final LoadingCache<Vault, VaultStatisticsComponent> vaultStats;
 	private final VaultStatisticsComponent.Builder vaultStatsBuilder;
+	private final BooleanProperty draggingUnlockedVaultContentOver = new SimpleBooleanProperty();
+
+	public Button dropZone;
 
 	@Inject
-	public VaultDetailUnlockedController(ObjectProperty<Vault> vault, FxApplicationWindows appWindows, VaultService vaultService, VaultStatisticsComponent.Builder vaultStatsBuilder, @MainWindow Stage mainWindow) {
+	public VaultDetailUnlockedController(ObjectProperty<Vault> vault, FxApplicationWindows appWindows, VaultService vaultService, VaultStatisticsComponent.Builder vaultStatsBuilder, @MainWindow Stage mainWindow, ResourceBundle resourceBundle) {
 		this.vault = vault;
 		this.appWindows = appWindows;
 		this.vaultService = vaultService;
 		this.mainWindow = mainWindow;
+		this.resourceBundle = resourceBundle;
 		this.vaultStats = CacheBuilder.newBuilder().weakValues().build(CacheLoader.from(this::buildVaultStats));
 		this.vaultStatsBuilder = vaultStatsBuilder;
 	}
 
+	public void initialize() {
+		dropZone.setOnDragEntered(this::handleDragEvent);
+		dropZone.setOnDragOver(this::handleDragEvent);
+		dropZone.setOnDragDropped(this::handleDragEvent);
+		dropZone.setOnDragExited(this::handleDragEvent);
+
+		EasyBind.includeWhen(dropZone.getStyleClass(), ACTIVE_CLASS, draggingUnlockedVaultContentOver);
+	}
+
+	private void handleDragEvent(DragEvent event) {
+		if (DragEvent.DRAG_OVER.equals(event.getEventType()) && event.getGestureSource() == null && event.getDragboard().hasFiles()) {
+			draggingUnlockedVaultContentOver.set(event.getDragboard().getFiles().stream().map(File::toPath).anyMatch(this::containsUnlockedVaultContent));
+			if (draggingUnlockedVaultContentOver.get()) {
+				event.acceptTransferModes(TransferMode.LINK);
+			}
+		} else if (DragEvent.DRAG_DROPPED.equals(event.getEventType()) && event.getGestureSource() == null && event.getDragboard().hasFiles()) {
+			Set<Path> ciphertextPaths = event.getDragboard().getFiles().stream().map(File::toPath).map(this::getCiphertextPath).flatMap(Optional::stream).collect(Collectors.toSet());
+			if (!ciphertextPaths.isEmpty()) {
+				ciphertextPaths.forEach(this::revealPath);
+			}
+			event.setDropCompleted(!ciphertextPaths.isEmpty());
```

Comment:
Additonally, we could copy the list of paths to the clipboard?

Even if there is no reveal implementation, the user would have the paths.

Reply:
Resolved in https://github.com/cryptomator/cryptomator/pull/2592/commits/e55d4dd11cf387a8ff9f48fbd394379987cce84a.

Only if no revealService is present, the paths are copied to clipboard